File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mpp;
        int n=nums.size();
        for(int i=0;i<n;i++){
            if(mpp.count(target-nums[i])){
                return {mpp[target-nums[i]],i};
            }
            mpp[nums[i]]=i;
        }
        return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer,Integer>mpp=new HashMap<>();
        int n=nums.length;

        for(int i=0;i<n;i++){
            int req=target-nums[i];
            if(mpp.containsKey(req)){
                return new int[]{mpp.get(req),i};
            }
            mpp.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}
